### PR TITLE
feat: Adds support for logo in portal settings page [CW-2585]

### PR DESCRIPTION
**Pull request by nithindavid**
## Description

Fixes # (issue)
https://linear.app/chatwoot/issue/CW-2585/help-center-support-logo-upload-for-the-header

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalSettingsBasicForm.vue [hunk 1]
```diff
@@ -159,13 +178,23 @@
         exampleURL: EXAMPLE_URL,
       });
     },
+    showDeleteButton() {
+      return this.logoUrl && !this.logoUrl.includes('www.gravatar.com');
```
[the remark this answers is not in the document]
Comment (scmmishra) [reply app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalSettingsBasicForm.vue:182]: Let's parse the URL and check the hostname explicitly

File: app/javascript/dashboard/routes/dashboard/settings/profile/Index.vue [hunk 2]
```diff
@@ -198,6 +198,9 @@ export default {
       currentUserId: 'getCurrentUserID',
       globalConfig: 'globalConfig/get',
     }),
+    showDeleteButton() {
+      return this.avatarUrl && !this.avatarUrl.includes('www.gravatar.com');
+    },
```
Comment (scmmishra) [inline app/javascript/dashboard/routes/dashboard/settings/profile/Index.vue:203]:
```suggestion
    showDeleteButton() {
      const logURLHost = new URL(this.logoUrl).host;
      const isFromGravatar = ['www.gravatar.com', 'gravatar'].includes(
        logURLHost
      );

      return this.logoUrl && !isFromGravatar;
    },
```

File: app/javascript/dashboard/routes/dashboard/settings/profile/Index.vue [hunk 3]
```diff
@@ -198,6 +198,9 @@ export default {
       currentUserId: 'getCurrentUserID',
       globalConfig: 'globalConfig/get',
     }),
+    showDeleteButton() {
+      return this.avatarUrl && !this.avatarUrl.includes('www.gravatar.com');
```
Comment (scmmishra) [inline app/javascript/dashboard/routes/dashboard/settings/profile/Index.vue:202]:
```suggestion
            const logURLHost = new URL(this.logoUrl).host;
      const isFromGravatar = ['www.gravatar.com', 'gravatar'].includes(
        logURLHost
      );

      return this.logoUrl && !isFromGravatar;
```

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalSettingsBasicForm.vue [hunk 4]
```diff
@@ -159,13 +178,23 @@ export default {
...
```
Comment (scmmishra) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalSettingsBasicForm.vue:182]:
```suggestion
      const logURLHost = new URL(this.logoUrl).host;
      const isFromGravatar = ['www.gravatar.com', 'gravatar'].includes(
        logURLHost
      );

      return this.logoUrl && !isFromGravatar;
```

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**